src/main/java/hudson/plugins/active_directory/ActiveDirectoryDomain.java

```diff
@@ -197,6 +204,13 @@ public TlsConfiguration getTlsConfiguration() {
         return tlsConfiguration;
     }
 
+    private Object readResolve() throws ObjectStreamException {
+        if (isFipsNonCompliant(tlsConfiguration != null && TlsConfiguration.TRUST_ALL_CERTIFICATES.name().equals(tlsConfiguration.name()))) {
```

Comment:
You could do the same enum `equals` check as in the constructor?

```suggestion
        if (isFipsNonCompliant(tlsConfiguration != null && TlsConfiguration.TRUST_ALL_CERTIFICATES.equals(tlsConfiguration))) {
```

Reply:
So of course ignore this if you go with my suggestion above to reuse the other static method :)